pkg/ccl/backupccl/restore_job.go

```diff
@@ -923,10 +936,57 @@ func createImportingDescriptors(
 				return errors.Wrapf(err, "restoring %d TableDescriptors from %d databases", len(r.tables), len(databases))
 			}
 
+			b := txn.NewBatch()
+
+			// For new schemas with existing parent databases, the schema map on the
+			// database descriptor needs to be updated.
+			existingDBsWithNewSchemas := make(map[descpb.ID][]catalog.SchemaDescriptor)
+			for _, sc := range writtenSchemas {
+				parentID := sc.GetParentID()
+				if _, ok := newDatabaseIDs[parentID]; !ok {
+					existingDBsWithNewSchemas[parentID] = append(existingDBsWithNewSchemas[parentID], sc)
+				}
+			}
+			// Write the updated databases.
+			for dbID, schemas := range existingDBsWithNewSchemas {
+				log.Infof(ctx, "writing schema entries to database %d", dbID)
+				desc, err := catalogkv.GetDescriptorByID(ctx, txn, keys.SystemSQLCodec,
+					dbID, catalogkv.Mutable, catalogkv.DatabaseDescriptorKind, true /* required */)
+				if err != nil {
+					return err
+				}
+				db := desc.(*dbdesc.Mutable)
+				if db.Schemas == nil {
+					// TODO (lucy): Push map initialization to when we create these
+					// descriptors.
+					db.Schemas = make(map[string]descpb.DatabaseDescriptor_SchemaInfo)
+				}
+				for _, sc := range schemas {
+					db.Schemas[sc.GetName()] = descpb.DatabaseDescriptor_SchemaInfo{ID: sc.GetID()}
+				}
+				// TODO (lucy): It's risky to be incrementing the version outside of the
```

Comment:
You could do this instead at `publishDescriptors` and use `PublishMultiple` to handle this. I don't think that we need these entries present until we actually turn the tables online

Reply:
Hmm, I'm trying to keep the schema and database descriptor writes in the same transaction. Maybe we should eventually use `PublishMultiple` (or whatever it becomes) at the start of the job too.